From 699f6e09d5f0505b539d86dfe8497cb542da0c6e Mon Sep 17 00:00:00 2001
From: Keith Mok <kmok@cyngn.com>
Date: Tue, 15 Dec 2015 13:24:34 -0800
Subject: [PATCH 03/16] sepolicy: Add domain for mkfs binaries

The init binary must transition to another domain when calling out to
executables. Create the mkfs domain for mkfs.f2fs such that init can
transition to it when formatting userdata/cache partitions if the
"formattable" flag is set.

Change-Id: I1046782386d171a59b1a3c5441ed265dc0824977
---
 common/private/file_contexts | 1 +
 common/private/mkfs.te       | 9 +++++++++
 2 files changed, 10 insertions(+)
 create mode 100644 common/private/mkfs.te

diff --git a/common/private/file_contexts b/common/private/file_contexts
index 3922553..164a682 100644
--- a/common/private/file_contexts
+++ b/common/private/file_contexts
@@ -1,6 +1,7 @@
 # Filesystem tools
 /system/bin/fsck\.exfat                 u:object_r:fsck_exec:s0
 /system/bin/fsck\.ntfs                  u:object_r:fsck_exec:s0
+/system/bin/mkfs\.f2fs                  u:object_r:mkfs_exec:s0
 
 # Fingerprint
 /(vendor|system/vendor)/bin/hw/android\.hardware\.biometrics\.fingerprint@2\.0-service u:object_r:hal_fingerprint_default_exec:s0
diff --git a/common/private/mkfs.te b/common/private/mkfs.te
new file mode 100644
index 0000000..fe7c61b
--- /dev/null
+++ b/common/private/mkfs.te
@@ -0,0 +1,9 @@
+type mkfs, domain;
+type mkfs_exec, exec_type, file_type;
+
+init_daemon_domain(mkfs)
+
+# Allow formatting userdata or cache partitions
+allow mkfs block_device:dir search;
+allow mkfs userdata_block_device:blk_file rw_file_perms;
+allow mkfs cache_block_device:blk_file rw_file_perms;
-- 
2.7.4

